vortex-array/src/compute/compare.rs

```diff
@@ -194,8 +196,8 @@ fn arrow_compare(
     operator: Operator,
 ) -> VortexResult<ArrayRef> {
     let nullable = left.dtype().is_nullable() || right.dtype().is_nullable();
-    let lhs = Datum::try_new(left.to_array())?;
-    let rhs = Datum::try_new(right.to_array())?;
+    let lhs = Datum::try_new(left.to_canonical()?.into_array())?;
+    let rhs = Datum::try_new(right.to_canonical()?.into_array())?;
```

Comment:
this is a bit cheeky but essentially if you want a custom arrow_compare behaviour you have to implement compare compute function and delegate yourself